qubesadmin/tools/__init__.py

```diff
@@ -162,6 +166,12 @@ def parse_qubes_app(self, parser, namespace):
                    vm.name not in namespace.exclude
             ]
         else:
+            def uuid_to_domain(uuid: str):
+                for domain in app.domains:
+                    if domain.uuid == uuid:
+                        return domain
```

Comment:
With https://github.com/QubesOS/qubes-core-admin/pull/574, the `uuid:...` syntax can be used as Admin API call target. This means, it should be relatively easy to get `app.domains["uuid:...."]` working, which is much better option than fetching UUID of all qubes and iterating client-side.
Technical hint: `get_blind()` won't work with just UUID, because the QubesVM object needs a name to be constructed. But that's okay, just skip the blind mode for getting VM via UUID.

Reply:
> This means, it should be relatively easy to get `app.domains["uuid:...."]` working

I had tried this before going to iteration path. It does not work.

> Technical hint: `get_blind()` won't work with just UUID

This actually works. But the QubesVM object is constructed with "uuid:..." for its name rather than the original vmname.

Reply:
@marmarek I just looked at `qubesadmin/app.py` on my system and recognized that it is v4.3.6. It is the latest [python3-qubesadmin-4.3.6-1.fc41.noarch.rpm](https://yum.qubes-os.org/r4.3/current-testing/dom0/fc41/rpm/python3-qubesadmin-4.3.6-1.fc41.noarch.rpm) in testing repositories. So the UUID changes I need is not yet released?

Reply:
UUID support is on the backend (core-admin, not core-admin-client). It's in qubes-core-dom0 4.3.9

Reply:
> UUID support is on the backend (core-admin, not core-admin-client). It's in qubes-core-dom0 4.3.9

Yes. That is correct. My mistake.

I gave up on adding UUID  support to VMCollection. I readjusted the PR, commit message and comments just for UNIX Wildcard support. It is still useful IMO. Users will be able to `qvm-kill disp*`, `qvm-start sys-*`, `qvm-run -p fedora* dnf upgrade -y --downloadonly` and similar tasks.